### ISSUE TITLE
Extension's V3 ServiceWorker

### ISSUE THREAD

**Issue by Alm0g** [wanderview/quota-storage-partitioning#14]
Hey :)
We have a Chrome extension that embeds an iframe into sites upon users clicking the extension icon.

That iframe used to share its login state with other iframe instances via Communication & Web Storage APIs, which are now partitioned.

Looking for a solution, I also read that ServiceWorkers spawned by 3Ps are also partitioned.

However, it looks like it doesn't affect the MV3 Chrome extension's ServiceWorker. 
So basically, this one is still a security hole, able to share different sites' data using a variety of storage mechanisms available in ServiceWorker's ecosystem.
Is this a conscious decision, or am I missing something?

Thanks!

**Comment by miketaylr** [comment]
If the extension has host permissions for the sites where an iframe is embedded, this is expected behavior. But it's a little hard to understand without knowing more about the chrome extension architecture and involved sites. If you still think this is a bug, would you mind filing an issue at crbug.com/new and providing a copy of the extension (if possible, if not, a detailed description of your setup).

**Comment by Alm0g** [comment]
@miketaylr Hi :)

Well: `host_permissions: ["https://*/*", "http://*/*"],`. So I guess this is the expected behavior.

I have to wonder what's the difference between an extension-backed iframe Versus just an iframe in terms of permissions. 
We allow unpartitioned capabilities for the former but not for the latter.

I mean, we could, potentially, have stored data from site A in `chrome.storage` (accessible through the service worker) and sent it to content scripts in other tabs, basically bypassing the partitioning update.

**Comment by miketaylr** [comment]
> I have to wonder what's the difference between an extension-backed iframe Versus just an iframe in terms of permissions.

The main difference is that a user (or their administrator) installed an extension, and in theory understands the implications of doing so vs clicking on a hyperlink and loading a random iframe in a page.

**Comment by Alm0g** [comment]
Thanks, it makes sense now :)
